sycl/include/CL/sycl/INTEL/esimd/esimd_view.hpp

```diff
@@ -24,6 +24,9 @@ namespace gpu {
 ///
 /// \ingroup sycl_esimd
 template <typename BaseTy, typename RegionTy> class simd_view {
+  template <typename, int> friend class simd;
+  template <typename, typename> friend class simd_view;
```

Comment:
Is this needed? Looks like you're trying to make a class be a friend of itself.

Reply:
Yes,  this is needed. Otherwise you can't access private members - such as private constructors - from member functions.
Note that this is not a class, this is templated class. Each instantiation generates unrelated class which are not friends to each other.

Reply:
I see. This is to allow accessing private constructors from different template instantiations of a class. Makes sense.

Reply:
Just in case, I would like to confirm this with C++/FE experts, since it caught your and @pvchupin 's eyes.

@erichkeane, can you please take a look if this friend template class declaration below is a good way to allow
calling private constructor from `select`:

```cpp
template <typename BaseTy, typename RegionTy> class simd_view {
  template <typename, typename> friend class simd_view; // <=== all simd_view instantiations are friends
  ...
private:
  simd_view(BaseTy &Base, RegionTy Region) : M_base(Base), M_region(Region) {}
  simd_view(BaseTy &&Base, RegionTy Region) : M_base(Base), M_region(Region) {}

public:
  template <int Size, int Stride, typename T = simd_view,
            typename = sycl::detail::enable_if_t<T::is1D()>>
  auto select(uint16_t Offset = 0) {
    using TopRegionTy = region1d_t<element_type, Size, Stride>;
    using NewRegionTy = std::pair<TopRegionTy, RegionTy>;
    using RetTy = simd_view<BaseTy, NewRegionTy>; // <======= call private constructor
    TopRegionTy TopReg(Offset);
    return RetTy{this->M_base, std::make_pair(TopReg, M_region)};
  }
  ...
}
```

Reply:
Yep, thats a pretty typical way to do that.  Note that the line you have labeled as 'call private constructor' isn't actually doing the call!  That is a 'using' statement.  It appears the 'return' line 2 lines lower is the one actually calling the private constructor.

Reply:
Thanks! Yes, I should've put the comment below to return of course.